Mathlib/Topology/Algebra/ProperAction/Basic.lean

```diff
@@ -208,3 +208,52 @@ instance [IsTopologicalGroup G] {H : Subgroup G} [H_closed : IsClosed (H : Set G
 instance QuotientGroup.instT2Space [IsTopologicalGroup G] {H : Subgroup G} [IsClosed (H : Set G)] :
     T2Space (G ⧸ H) :=
   t2Space_quotient_mulAction_of_properSMul
+
+/-- If `G` acts on `X` properly, then the map `G × T → X × T, (g, t) ↦ (g • t, t)` is still
+proper for *any* subset `T` of `X`. -/
+@[to_additive
+/-- If `G` acts on `X` properly, then the map `G × T → X × T, (g, t) ↦ (g +ᵥ t, t)` is still
+proper for *any* subset `T` of `X`. -/]
+lemma ProperSMul.isProperMap_smul_pair_set [ProperSMul G X] {t : Set X} :
+    IsProperMap (fun (gx : G × t) ↦ ((gx.1 • gx.2, gx.2) : X × t)) := by
+  let Φ : G × X → X × X := fun gx ↦ (gx.1 • gx.2, gx.2)
```

Comment:
Style nit:
```suggestion
    IsProperMap (fun (⟨g, x⟩ : G × t) ↦ ((g • x, x) : X × t)) := by
  let Φ : G × X → X × X := fun ⟨g, x⟩ ↦ (g • x, x)
```

Reply:
@grunweg It is, otherwise Lean tries to find an action of `G` on `t`.

@ocfnash The reason I usually avoid these (at least in statements) is because the actual expression is more complex (and ugly, I don't want match statements in my infoview). I am actually surprised that `fun_prop` works with such functions, so maybe it's not as bad as I thought, but I am scared about the way it will be displayed in the documentation... In any case, I think I would prefer staying consistent with https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/Algebra/ProperAction/Basic.html#ProperSMul.isProperMap_smul_pair, and maybe switch later if we decide so?

Reply:
Oh yes, fair enough. Ignore this suggestion then: we need to lobby core Lean to solve this!